### PR TITLE
feat(number-field): add new event up-click and down-click

### PR DESCRIPTION
**Pull request by Theeraphat-Sorasetsakul**
## Description

- Add new event named up-click and down-click.
- Calls when tap at spinner.
- if do preventDefault, value-changed event should not call.

Fixes # ([ELF-2222](https://jira.refinitiv.com/browse/ELF-2222))

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/elements/src/number-field/__test__/number-field.test.js [hunk 1]
```diff
@@ -306,6 +306,101 @@ describe('number-field/NumberField', function () {
       await elementUpdated(el);
       expect(el.value).to.equal('0', 'value should not be less then zero');
     });
+    it('Should fires up-click and value-changed when keydown spinner up', async function () {
```
Comment (Nantawat-Poothong) [inline packages/elements/src/number-field/__test__/number-field.test.js]: I got confusing between `up-click` and `value-changed`, their are similar behavior but different naming pattern 😵‍💫

Comment (Nantawat-Poothong) [inline packages/elements/src/number-field/__test__/number-field.test.js]:
```suggestion
    it('Arrow up should fires up-click and value-changed ', async function () {
```

File: packages/elements/src/number-field/__test__/number-field.test.js [hunk 2]
```diff
@@ -306,6 +306,101 @@ describe('number-field/NumberField', function () {
...
+        valueChangedCount += 1;
+      });
+      setTimeout(() => dispatchTapEvent(spinnerDownEl));
+      await oneEvent(spinnerDownEl, 'tap');
+      await elementUpdated(el);
+      expect(el.value).to.equal('-1');
+      expect(downClickedCount).to.equal(1);
+      expect(valueChangedCount).to.equal(1);
+    });
+    it('Should stop view-changed event when set prevent default in up-click', async function () {
```
Comment (Nantawat-Poothong) [inline packages/elements/src/number-field/__test__/number-field.test.js]:
```suggestion
    it('Should stop value-changed event when set prevent default in up-click', async function () {
```

File: packages/elements/src/number-field/__test__/number-field.test.js [hunk 3]
```diff
@@ -306,6 +306,101 @@ describe('number-field/NumberField', function () {
...
+      el.addEventListener('value-changed', () => {
+        valueChangedCount += 1;
+      });
+      setTimeout(() => dispatchTapEvent(spinnerUpEl));
+      await oneEvent(spinnerUpEl, 'tap');
+      await elementUpdated(el);
+      expect(el.value).to.equal(value, 'Should not update value if up-click does prevent default');
+      expect(valueChangedCount).to.equal(0, 'Should not call value-changed if up-click does prevent default');
+    });
+    it('Should stop view-changed event when set prevent default in down-click', async function () {
```
Comment (Nantawat-Poothong) [inline packages/elements/src/number-field/__test__/number-field.test.js]:
```suggestion
    it('Should stop value-changed event when set prevent default in down-click', async function () {
```

File: packages/elements/src/number-field/__test__/number-field.test.js (hunk 2 above)
Comment (Nantawat-Poothong) [reply packages/elements/src/number-field/__test__/number-field.test.js]: or how about `value-changed event should not fires when set prevent default to up-click`?

File: packages/elements/src/number-field/__test__/number-field.test.js (hunk 3 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/number-field/__test__/number-field.test.js]: Updated by following your rephase suggestion.

File: packages/elements/src/number-field/__test__/number-field.test.js (hunk 2 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/number-field/__test__/number-field.test.js]: Updated by following your rephase suggestion.

**Review by Nantawat-Poothong — DISMISSED** [review]
LGTM

**Review by Nantawat-Poothong — APPROVED** [review]
LGTM

**Comment by bualoy-napat** [comment]
LGTM

**Pull request merged**